pkg/services/ngalert/api/api_ruler.go

```diff
@@ -636,5 +637,50 @@ func calculateChanges(ctx context.Context, ruleStore store.RuleStore, groupKey n
 	}, nil
 }
 
+// calculateAutomaticChanges scans all affected groups and creates either a noop update that will increment the version of each rule as well as re-index other groups.
```

Comment:
If no groups at all are affected, does the version still increment?

Reply:
if no group are affected i.e. no changes, version does not increment.